packages/circuit-ui/components/ColorInput/ColorInput.tsx

```diff
@@ -0,0 +1,142 @@
+/**
+ * Copyright 2024, SumUp Ltd.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use client';
+
+import {
+  forwardRef,
+  useCallback,
+  useEffect,
+  useId,
+  useRef,
+  useState,
+  type ChangeEventHandler,
+} from 'react';
+
+import { Input, type InputElement, type InputProps } from '../Input/index.js';
+import { clsx } from '../../styles/clsx.js';
+import { FieldLabel, FieldLabelText } from '../Field/index.js';
+import { applyMultipleRefs } from '../../util/refs.js';
+
+import styles from './ColorInput.module.css';
+
+export interface ColorInputProps
+  extends Omit<
+    InputProps,
+    | 'ref'
+    | 'type'
+    | 'defaultValue'
+    | 'value'
+    | 'placeholder'
```

Comment:
What's the reasoning behind omitting the props here and redeclaring them below? 🤔

Reply:
For most of them it is to restrict the types further. E.g. for `defaultValue`, `value`, `placeholder`, etc. string is the only valid type (as compared to allowing `number` as well for example).